### DISCUSSION TITLE
Server empty for 60 seconds, pausing Feature.

### DISCUSSION THREAD

**Discussion by abhiyanpa** [PaperMC/Paper#11571] [Ideas]
### Is your feature request related to a problem?

It would  be better if you add this feature so we can manage our resources efficiently.

### Describe the solution you'd like.

Add an option to automatically pause the server when no players are connected for over 60 seconds. This functionality would save resources, reduce CPU usage, and potentially decrease hosting costs for smaller servers or development environments.

### Describe alternatives you've considered.

Using External Scripts or Tools
I’ve tried using external scripts to monitor player activity and issue a command to pause or shut down the server when it’s empty. However, this approach adds complexity and can lead to resource overhead. Since it operates outside PaperMC, there’s also a slight delay in detecting when a player joins again, which impacts the experience.

Scheduled Restarts and Downtime
I considered setting up scheduled server restarts or downtimes, especially during late hours when player traffic is low. While this helps save resources, it’s not ideal because it doesn’t adapt to real-time player activity, meaning players may find the server offline even if they try to join outside peak hours.

Custom Plugin Development
I looked into creating a custom plugin to handle pausing when the server is empty. However, developing and maintaining a custom plugin requires significant time and ongoing updates to remain compatible with PaperMC changes. I believe a built-in feature would be more efficient and accessible for the community.

**Comment by electronicboy** [comment]
There is no real concept of pausing a running process from inside the process, best you could do is maybe reduce how much you spin wait, but, stuff still generally expects to run even when the server is empty. This is generally not worth the sunken cost of trying to implement this properly, IMHO, you're still running an entire JVM with all of the other bits of software like plugins in there which expect to be able to do things.

**Reply by abhiyanpa** [reply]
Even Forge that has feature i can attach a screenshot
.
<img width="345" alt="{1A189C2C-1533-4A46-BCE6-20AC464A7444}" src="https://github.com/user-attachments/assets/ce545d6b-4308-49a5-b4e6-61d2910a9a19">

**Reply by electronicboy** [reply]
That's a vanilla feature

**Reply by abhiyanpa** [reply]
Yes but why Paper isn't using it? it's better thou

**Reply by masmc05** [reply]
Paper disables it by default since it can break plugins, you can enable it in the same place as everywhere else, in `server.proprietes`

**Reply by electronicboy** [reply]
You can enable it in server.properties, paper disables it by default due to some concerns over how spigot implemented this feature into their own codebase, i.e. not allowing the chunk system to unload chunks. Some of this has been mitigated, but, you'll need to test it yourself and see if it works for your setup.
